pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -890,3 +891,421 @@ func TestVolumeBinding(t *testing.T) {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cap-a",
+				},
+				StorageClassName: "sc-b",
+			},
+			newCap: &storagev1.CSIStorageCapacity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cap-a",
+				},
+				StorageClassName: "sc-b",
+			},
+			pvcLister: tf.PersistentVolumeClaimLister{
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-a", "sc-a").PersistentVolumeClaim
+					return *pvc
+				}(),
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-b", "sc-b").PersistentVolumeClaim
+					return *pvc
+				}(),
+			},
+			err:    false,
+			expect: framework.QueueSkip,
+		},
+		{
+			name: "skipping the check for CSIStorageCapacity as the pvc is configured to be bound to an existing pv",
+			pod: makePod("pod-a").
+				withPVCVolume("pvc-a", "").
+				withPVCVolume("pvc-b", "").
+				Pod,
+			oldCap: nil,
+			newCap: &storagev1.CSIStorageCapacity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cap-a",
+				},
+				StorageClassName: "sc-b",
+			},
+			pvcLister: tf.PersistentVolumeClaimLister{
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-a", "sc-a").PersistentVolumeClaim
+					return *pvc
+				}(),
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-b", "sc-b").withBoundPV("pv-b").PersistentVolumeClaim
+					return *pvc
+				}(),
+			},
+			err:    false,
+			expect: framework.QueueSkip,
+		},
+		{
+			name: "the storage class of the specified by pvc is not found",
+			pod: makePod("pod-a").
+				withPVCVolume("pvc-a", "").
+				withPVCVolume("pvc-b", "").
+				Pod,
+			oldCap: &storagev1.CSIStorageCapacity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sc-a",
+				},
+			},
+			newCap: &storagev1.CSIStorageCapacity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sc-a",
+				},
+			},
+			pvcLister: tf.PersistentVolumeClaimLister{
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-a", "sc-a").PersistentVolumeClaim
+					return *pvc
+				}(),
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-b", "sc-notfound").PersistentVolumeClaim
+					return *pvc
+				}(),
+			},
+			err:    false,
+			expect: framework.QueueSkip,
+		},
+		{
+			name: "pod has pvc that references a newly added CSIStorageCapacity",
+			pod: makePod("pod-a").
+				withPVCVolume("pvc-a", "").
+				withPVCVolume("pvc-b", "").
+				Pod,
+			oldCap: nil,
+			newCap: &storagev1.CSIStorageCapacity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cap-a",
+				},
+				StorageClassName: "sc-b",
+			},
+			pvcLister: tf.PersistentVolumeClaimLister{
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-a", "sc-a").PersistentVolumeClaim
+					return *pvc
+				}(),
+				func() v1.PersistentVolumeClaim {
+					pvc := makePVC("pvc-b", "sc-b").PersistentVolumeClaim
+					return *pvc
+				}(),
+			},
+			err:    false,
+			expect: framework.Queue,
+		},
+		{
+			name: "pod has ephemeral volume that references a newly added CSIStorageCapacity",
+			pod: func() *v1.Pod {
```

Comment:
Can we extend the functionality of `withGenericEphemeralVolume` or add a new method to support this use case?

Reply:
I think it's a good idea, but I believe it might be smoother to address this after some PRs related to volume binding have been merged.
What are your thoughts on this approach?

Reply:
I think it's fine to address it with a follow-up PR. In that case, I thought about writing a TODO comment to avoid being pointed out by other reviewers, but I'll leave it up to you.